src/cljc/matcher_combinators/core.cljc

```diff
@@ -268,41 +268,91 @@
           (match expected transformed))))
     (-base-name [_] (-base-name expected)))
 
+(defn insert-all-combinations
+  "List of all ways of inserting elements from `insert-list` into `base-list`
+  while preserving the ordering that `base-list` elements have with respect to
+  each other"
+  [base-list insert-list]
+  (let [insert (fn [v i e] (vec (concat (take i v) [e] (drop i v))))
+        insert-count (count insert-list)
+        insert-indice-sets (reverse (combo/combinations (range (+ (count base-list) insert-count))
+                                                        insert-count))]
+    (map #(first (reduce (fn [[result-list remaining-insert-elems] index]
+                           [(insert result-list index (first remaining-insert-elems))
+                            (rest remaining-insert-elems)])
+                           [base-list insert-list]
+                           %))
+         insert-indice-sets)))
+
+#_(insert-all-combinations [1 2 3] [0 0])
+
 (defn- normalize-inputs-length
   "Modify the matchers and actuals sequences to match in length.
-  When `matchers` is longer, add `missing` elements to `actuals`.
-  When `actuals` is longer, add unexpected entry matchers to `matchers`."
+
+   - When `matchers` is longer, return pairs of `matchers` and all combinations
+     of `missing` inserted into `actuals`.
+   - When `actuals` is longer, return pairs of `actuals` and all combinations
+     of `unexpected` inserted into `matchers`.
+
+    The all-combinations results help compute minimized mismatches for sequences"
   [matchers actuals]
   (let [matchers-count (count matchers)
         actuals-count  (count actuals)]
-    (if (< actuals-count matchers-count)
-      [matchers
-       (take matchers-count (concat actuals (repeat ::missing)))]
-      [(take actuals-count (concat matchers (repeat unexpected-matcher)))
-       actuals])))
+    (cond (= actuals-count matchers-count)
+          [[matchers actuals]]
+
+          (< actuals-count matchers-count)
+          (map vector
+               (repeat matchers)
+               (insert-all-combinations actuals
+                                        (repeat (- matchers-count (count actuals))
+                                                ::missing)))
```

Comment:
This might make computing mismatches for sequences a slower.
One way to deal with it would be to create a dedicated matcher that enables this logic. I don't like this because it creates more options for users and this new mismatch behavior is probably something people want by default.

Another way to limit how much it could potentially slow things down would be to cap how many combinations are considered. Something like a `(take 30 ..)` around the `(insert-all-combinations ..)` thing here. (probably bound to a dynamic var for configuration)